pages/code_schools.js

```diff
@@ -1,71 +1,172 @@
-import Head from 'components/head';
+import Select from 'react-select';
 import { getCodeSchoolsPromise } from 'common/constants/api';
+import Head from 'components/head';
+import HeroBanner from 'components/_common_/HeroBanner/HeroBanner';
 import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
 import Section from 'components/_common_/Section/Section';
+import Button from 'components/_common_/Button/Button';
 import SchoolCard from 'components/Cards/SchoolCard/SchoolCard';
 import { s3 } from 'common/constants/urls';
+import States from 'common/constants/dropdown-states-values';
+import styles from './styles/code_schools.css';
 
 export default class CodeSchools extends React.Component {
   state = {
-    schools: [],
+    allSchools: [],
+    filteredSchools: [],
+    selectedStates: [],
   };
 
   async componentDidMount() {
     const { data } = await getCodeSchoolsPromise();
-    this.setState({ schools: data });
+    this.setState({ allSchools: data });
+    this.setState({ filteredSchools: data });
   }
 
+  filterVaApproved = () => {
+    const { allSchools } = this.state;
+    const vaApproved = allSchools.filter(school => {
+      const vaFound = school.locations.some(location => location.va_accepted === true);
+      return vaFound;
+    });
+    this.setState({ filteredSchools: vaApproved });
+    this.setState({ selectedStates: [] });
+  };
+
+  filterOnline = () => {
+    const { allSchools } = this.state;
+    const onlineSchools = allSchools.filter(school => school.has_online);
+    this.setState({ filteredSchools: onlineSchools });
+    this.setState({ selectedStates: [] });
+  };
+
+  filterMoocs = () => {
```

Comment:
Can you remove this button and logic? The mock doesn't appear to want to filter by MOOCs

Reply:
For me it looked like that Mooc is just like any other online only school. I personally see no reason to separate them into special section. But I guess we could stick them to the bottom of the page like in the old implementation.

Reply:
It's just about following the mock that design gave at this point. We'll implement analytics and find out if they're suffering. MOOCs are also very distinctly different than the other coding schools and wont actually be listed with the Code School card.

Reply:
Sounds good, will create special mooc card with logo and link button. Not sure where the mooc description is coming from. There is no such filed on incoming json.

Reply:
I'd like to reuse SuccessStory card for mooc school display. I guess there is another story to redesign it:  https://github.com/OperationCode/front-end/issues/291

Reply:
Indeed! Good catch. Your more than welcome to pause here, and finish out that issue.